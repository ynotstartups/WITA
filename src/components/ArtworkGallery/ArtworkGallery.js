import React from "react"
import Gallery from "react-photo-gallery"
import PropTypes from "prop-types"
import { withTheme } from "@material-ui/core/styles"

const ArtworkGallery = ({ theme, photos, id }) => {
  const { sm, md, lg } = theme.breakpoints.values
  function columns(containerWidth) {
    let columnsNumber = 1
    if (containerWidth >= sm) columnsNumber = 2
    if (containerWidth >= md) columnsNumber = 3
    if (containerWidth >= lg) columnsNumber = 4
    return columnsNumber
  }

  return (
    <Gallery photos={photos} direction="column" columns={columns} margin={16} />
  )
}

export default withTheme(ArtworkGallery)

ArtworkGallery.propTypes = {
  photos: PropTypes.arrayOf(
    PropTypes.shape({
      src: PropTypes.string.isRequired,
      width: PropTypes.number.isRequired,
      height: PropTypes.number.isRequired,
    })
  ).isRequired,
  theme: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
}
