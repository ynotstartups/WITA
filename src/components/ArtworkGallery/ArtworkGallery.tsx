import React from "react"
import Gallery from "react-photo-gallery"
import { withTheme } from "@material-ui/core/styles"

import Artwork, { Photo } from "./Artwork/Artwork"

interface Props {
  theme: any
  photos: Photo[]
}

const ArtworkGallery: React.FunctionComponent<Props> = ({ theme, photos }) => {
  const { sm, md, lg } = theme.breakpoints.values
  function columns(containerWidth) {
    let columnsNumber = 1
    if (containerWidth >= sm) columnsNumber = 2
    if (containerWidth >= md) columnsNumber = 3
    if (containerWidth >= lg) columnsNumber = 4
    return columnsNumber
  }

  return (
    <Gallery
      photos={photos}
      direction="column"
      columns={columns}
      margin={0}
      // @ts-ignore
      renderImage={Artwork}
    />
  )
}

export default withTheme(ArtworkGallery)
