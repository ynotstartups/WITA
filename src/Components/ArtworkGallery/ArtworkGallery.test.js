import React from "react";
import renderer from "react-test-renderer";
import ArtworkGallery from "./ArtworkGallery";

it("renders correctly", () => {
  const theme = {
    breakpoints: {
      values: {
        sm: 1,
        md: 2,
        lg: 3
      }
    }
  };
  const tree = renderer
    .create(<ArtworkGallery theme={theme} photos={[]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
