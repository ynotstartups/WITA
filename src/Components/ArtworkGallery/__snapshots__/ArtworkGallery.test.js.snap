// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkGallery renders correctly 1`] = `
<div>
  Â 
</div>
`;
